### ISSUE TITLE
[CT-2688] [Regression] Can no longer target analysis in a selection

### ISSUE THREAD

**Issue by moltar** [dbt-labs/dbt-core#7849]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

This no longer works in 1.5.1:

```sh
dbt compile --select "analysis/"
```

However, if I run without the `--select`:

```sh
dbt compile
```

The analysis are found and are compiled.

### Expected Behavior

To support the selection of `analysis` as a path.

### Steps To Reproduce

Run:

```sh
dbt compile --select "analysis/"
```

Observe that `target/*/analysis` is empty.

### Relevant log output

```shell
N/A
```

### Environment

```markdown
- OS: macOS
- Python: 3.11
- dbt: 1.5.1
```

### Which database adapter are you using with dbt?

postgres

### Additional Context

This selection method worked from at least 1.3.0 to 1.4.6.

**Comment by jtcohen6** [comment]
@moltar I'm not able to reproduce this locally:
```yml
# dbt_project.yml
name: ...
profile: ...
analysis-paths: ["analysis"]
```

```sql
-- analysis/my_analysis.sql
-- {{ run_started_at }}
select 1 as id
```

```bash
$ dbt compile -s "analysis/"
14:04:10  Running with dbt=1.5.1
14:04:10  Found 1 model, 0 tests, 0 snapshots, 1 analysis, 307 macros, 0 operations, 0 seed files, 0 sources, 0 exposures, 0 metrics, 1 group
14:04:10
14:04:11  Concurrency: 5 threads (target='dev')
14:04:11
```

```
-- target/compiled/my_dbt_project/analysis/my_analysis.sql
-- 2023-06-13 14:04:10.765988+00:00
select 1 as id
```

**Comment by gshank** [comment]
Were you using a project_dir that's not the root directory of your project? This might be another example of #7819.

**Comment by moltar** [comment]
> Were you using a project_dir that's not the root directory of your project? 

Yes!! 😁 

```sh
export DBT_PROJECT_DIR="sub/dir/dbt"
export DBT_PROFILES_DIR="$DBT_PROJECT_DIR"

dbt compile --select "analysis/"
```

**Comment by dbeatty10** [comment]
@moltar I re-created a dbt project with environment variables similar to #7819, and all of the following worked for me when using dbt~=1.5.0 if I removed the trailing slash:
```shell
dbt compile --select analysis
```
and
```shell
dbt compile --select analysis --project-dir $DBT_PROJECT_DIR
```
and
```shell
dbt compile --select analysis --project-dir $DBT_PROJECT_DIR --profiles-dir $DBT_PROFILES_DIR
```

But using a trailing slash didn't work for me in dbt v1.4 or v1.5 🤷 (Using double quotes or not didn't affect my results.)

Could you give it a shot without the trailing slash and see if it works or not?

**Comment by moltar** [comment]
Hey @dbeatty10, thanks for the tip! No slash does indeed work. 🎉 

I thought I'd tried everything, including that 😁 

Other things I tried were using `path:` prefix, full paths from the root, etc...

Btw, for posterity, if anyone comes across this, one other thing needed in 1.5, for those who do not have dbt in the root:

```sh
export DBT_TARGET_PATH="$DBT_PROFILES_DIR/target"
```

This is due to a bug.

**Comment by dbeatty10** [comment]
> Hey @dbeatty10, thanks for the tip! No slash does indeed work. 🎉

Awesome, great news @moltar 🎉

> This is due to a bug

Is https://github.com/dbt-labs/dbt-core/issues/7465 the bug you are referring to?

If so, you can try out the [1.5.2rc1](https://pypi.org/project/dbt-core/1.5.2rc1/) release candidate which should include a fix for that issue.

Assuming we have everything covered, I'm going to close this issue as resolved. But please let me know if not, and I'll be happy to re-open.

**Comment by moltar** [comment]
> Is #7465 the bug you are referring to?

Yes, that's the one.

> Assuming we have everything covered

So, is _not_ including the slash the solution?

If so, to be clear, dbt used to work with the slash, wondering if this would be a regression for others.

Or is the fix for no slash is part of the [1.5.2rc1](https://pypi.org/project/dbt-core/1.5.2rc1/) release too?

Thanks.
